diff --git a/auth/middleware.go b/auth/middleware.go
index 033c489..91a6904 100644
--- a/auth/middleware.go
+++ b/auth/middleware.go
@@ -258,16 +258,17 @@ func internalAuth(internalNet []*net.IPNet, payload []byte,
 	if ip == nil {
 		panic(fmt.Errorf("Unable to parse remote address"))
 	}
-	var ok bool = false
-	for _, ipnet := range internalNet {
-		ok = ok || ipnet.Contains(ip)
-		if ok {
-			break
+	var route []string
+	for _, val := range r.Header.Values("X-Forwarded-For") {
+		route = append(route, strings.Split(val, ",")...)
+	}
+	for _, val := range route {
+		routeIP := net.ParseIP(val)
+		if routeIP == nil {
+			continue
 		}
-	}
-	if !ok {
-		authError(w, fmt.Sprintf("Invalid source IP %s for internal auth", ip), http.StatusUnauthorized)
-		return
+		ip = routeIP
+		break
 	}
 
 	payload = crypto.DecryptWithExpiration(payload, 30*time.Second)
